### ISSUE TITLE
php56: configure: error: DBA: Could not find necessary header file(s).

### ISSUE THREAD

**Issue by aiyuchen** [Linuxbrew/legacy-linuxbrew#360]
```
brew install -vd homebrew/php/php56 --homebrew-apxs --without-apache --with-debug --with-fpm --with-gmp --with-homebrew-curl --with-homebrew-libxslt --with-homebrew-openssl --with-libmysql --with-tidy
```

gits-logs: https://gist.github.com/anonymous/ac9726a67198e4c0a946

I have install `db4-devel.x86_64`

```
sudo yum install db4-devel
```

and I have install `gdbm` by linuxbrew

and search Issue like this
- https://github.com/Homebrew/homebrew-php/issues/1416
- https://github.com/Homebrew/homebrew-php/issues/1396#issuecomment-59987249
- https://github.com/Homebrew/homebrew-php/issues/309

but `xcode-select` is not CentOS command

How can I do it?

**Comment by aiyuchen** [comment]
@sjackman

**Comment by aiyuchen** [comment]
@mistydemeo @adamv @jacknagel @xu-cheng @mikemcquaid @asparagui @tdsmith @mxcl

**Comment by jacknagel** [comment]
Please do not do that. It will not get you help faster.

**Comment by sjackman** [comment]
Try `brew install berkeley-db4`

**Comment by RWOverdijk** [comment]
What's the fix for this? @sjackman his suggestion doesn't work for me.

```
~ [ brew reinstall --with-homebrew-curl --with-homebrew-openssl --without-snmp --with-apache php56                                                 ] 11:40 AM
==> Reinstalling homebrew/php/php56 with --with-homebrew-curl
==> Downloading https://php.net/get/php-5.6.10.tar.bz2/from/this/mirror
Already downloaded: /Library/Caches/Homebrew/php56-5.6.10
Warning: Backing up all known pear.conf and .pearrc files
Warning: If you have a pre-existing pear install outside
         of homebrew-php, or you are using a non-standard
         pear.conf location, installation may fail.
==> ./configure --prefix=/usr/local/Cellar/php56/5.6.10 --localstatedir=/usr/local/var --sysconfdir=/usr/local/etc/php/5.6 --with-config-file-path=/usr/local/
checking for curl_easy_strerror in -lcurl... yes
checking for curl_multi_strerror in -lcurl... yes
checking for QDBM support... no
checking for GDBM support... no
configure: error: DBA: Could not find necessary header file(s).
```

**Comment by sjackman** [comment]
Please report the output of
```sh
brew install berkeley-db4
brew install --with-homebrew-curl --with-homebrew-openssl --without-snmp --with-apache php56
brew gist-logs php56
```

**Comment by stanley-cheung** [comment]
I run into the same issue and here's the output for the above commands:

https://gist.github.com/57be3c49db31cf88abac

**Comment by sjackman** [comment]
Try `brew install gdbm`

**Comment by stanley-cheung** [comment]
Same error
```
$ brew install gdbm
==> Downloading http://ftpmirror.gnu.org/gdbm/gdbm-1.11.tar.gz
==> Downloading from http://gnu.opencube.ca/gdbm/gdbm-1.11.tar.gz
######################################################################## 100.0%
==> ./configure --disable-silent-rules --prefix=/home/stanleycheung/.linuxbrew/Cellar/gdbm/1.11
==> make install
==> Caveats

==> Summary
/home/stanleycheung/.linuxbrew/Cellar/gdbm/1.11: 27 files, 828K, built in 6 seconds
stanleycheung@stanleycheung-test:~$ brew install php56
==> Installing php56 from homebrew/homebrew-php
==> Downloading https://php.net/get/php-5.6.11.tar.bz2/from/this/mirror
Already downloaded: /home/stanleycheung/.cache/Homebrew/php56-5.6.11
==> ./configure --prefix=/home/stanleycheung/.linuxbrew/Cellar/php56/5.6.11_2 --localstatedir=/home/stanleycheung/.linuxbrew/var --sysconfdir=/home/stanle
checking for curl_easy_strerror in -lcurl... yes
checking for curl_multi_strerror in -lcurl... yes
checking for QDBM support... no
checking for GDBM support... no
configure: error: DBA: Could not find necessary header file(s).
```
I also already have `libgdbm3`, `libgdbm-dev`, `libqdbm-dev` installed via apt-get

**Comment by stanley-cheung** [comment]
Still the same even if I `brew install qdbm`

**Comment by tdsmith** [comment]
The php formula passes --with-ndbm=/usr: https://github.com/Homebrew/homebrew-php/blob/0b56488efe2ef3f03c0319b95aa08adcee58665c/Abstract/abstract-php.rb#L212

That probably isn't appropriate for Linuxbrew.

**Comment by sjackman** [comment]
Can `php56` use `berkeley-db` rather than `berkeley-db4`?

**Comment by sjackman** [comment]
Try this patch. It can be improved, but is a quick first pass.
```diff
diff --git a/Abstract/abstract-php.rb b/Abstract/abstract-php.rb
index 5f70856..1c213e2 100644
--- a/Abstract/abstract-php.rb
+++ b/Abstract/abstract-php.rb
@@ -203,13 +203,13 @@ INFO
       "--with-freetype-dir=#{Formula['freetype'].opt_prefix}",
       "--with-gd",
       "--with-gettext=#{Formula['gettext'].opt_prefix}",
-      "--with-iconv-dir=/usr",
+      ("--with-iconv-dir=/usr" if OS.mac?),
       "--with-icu-dir=#{Formula['icu4c'].opt_prefix}",
       "--with-jpeg-dir=#{Formula['jpeg'].opt_prefix}",
-      "--with-kerberos=/usr",
+      ("--with-kerberos=/usr" if OS.mac?),
       "--with-libedit",
       "--with-mhash",
-      "--with-ndbm=/usr",
+      ("--with-ndbm=/usr" if OS.mac?),
       "--with-pdo-odbc=unixODBC,#{Formula['unixodbc'].opt_prefix}",
       "--with-png-dir=#{Formula['libpng'].opt_prefix}",
       "--with-unixODBC=#{Formula['unixodbc'].opt_prefix}",
@@ -229,7 +229,7 @@ INFO
     end
 
     if build.with? 'bz2'
-      args << '--with-bz2=/usr'
+      args << '--with-bz2=/usr' if OS.mac?
     end
 
     if build.with? 'debug'
@@ -274,7 +274,7 @@ INFO
     if build.with? 'homebrew-libxslt'
       args << "--with-xsl=" + Formula['libxslt'].opt_prefix.to_s
     else
-      args << "--with-xsl=/usr"
+      args << "--with-xsl=/usr" if OS.mac?
     end
 
     if build.with? 'imap'
```

**Comment by stanley-cheung** [comment]
Thanks for the patch!

I tried this patch, the install of `php56` did get past the original error above, then it failed with this error

`configure: error: Cannot find ldap libraries in /usr/lib.`

Then I try `brew install --without-apache --without-ldap php56`, which results in

`configure: error: Please reinstall libedit - I cannot find readline.h`

I do have both `libedit` and `readline` installed

```
$ dpkg -l | grep libedit
ii  libedit2:amd64                 3.1-20140620-2               amd64        BSD editline and history libraries
$ brew list | grep readline
readline
```

I tried to re-install those, tried a bunch of combination of `CFLAGS` but cannot get past this error.

So I guess `homebrew-php` just isn't well supported under `linuxbrew`?

**Comment by sjackman** [comment]
> So I guess homebrew-php just isn't well supported under linuxbrew?

True. It's not. It's up to you whether you think it's worth the effort to get it working.

Try `brew install homebrew/dupes/libedit`

**Comment by hannwong** [comment]
@stanley-cheung Same here. I reinstalled libedit from brew, and also installed ``libreadline-dev`` from apt-get. Still the same error.

The part of the error you didn't state is that PHP's ``configure`` found ``libedit`` (a replacement for readline), and so tries to find the ``readline.h`` for ``libedit``.

For the record, I have tried all scenarios with ``readline.h``. Whether it is installed through apt-get or through Linuxbrew, and whether it be libedit or plain readline, the same error still comes up.

This is taking too long for us to fix. At this point, it is easier to just drop Linuxbrew and go with apt-get. It is also easier to do a compile from source, in which case we could be debugging plain ``configure`` scripts rather than trying to trace through Linuxbrew.

Note to stanley and our own teams: We're deciding to let our Mac users continue to use Homebrew (which works without a hitch), and let our Ubuntu users use apt-get. For those of us who need the latest versions of software, we will help them do a compile from source. (In fact, creating Debian binary packages is easier for us to do than working through Linuxbrew!)

In short... apt-get packages _are_ maintained, and so are Homebrew packages. Linuxbrew packages are not even tested to begin with, let alone maintained.

@sjackman We're willing to help you fix up Linuxbrew if you are willing to help us get started. Tell us where you left off so we know where to get started from. For example, you could tell us which packages _are_ tested, so we know to work on those that are not yet tested (such as homebrew/php/php56).

**Comment by sjackman** [comment]
I believe `homebrew/php/php56` depends on Apache. Here's my progress so far with installing `brew/apache/httpd24`.

```sh
❯❯❯ brew tap homebrew/apache
❯❯❯ brew install httpd24
Error: Operation already in progress for zlib
Another active Homebrew process is already using zlib.

❯❯❯ brew edit httpd24
# Change "homebrew/dupes/zlib" to "zlib"

❯❯❯ brew install postgresql
configure: error: library 'xslt' is required for XSLT support

# postgresql depends on libxslt
❯❯❯ brew install libxslt

❯❯❯ brew install postgresql
configure: error: library 'uuid' is required for E2FS UUID

# postgresql depends on libuuid
# Install e2fsprogs for libuuid
❯❯❯ brew install e2fsprogs
# e2fsprogs is keg-only and must be linked with --force
❯❯❯ brew link --force e2fsprogs

❯❯❯ brew install postgresql
configure: error: file 'tclConfig.sh' is required for Tcl

# postgresql depends on tcl
❯❯❯ brew install tcl-tk

❯❯❯ brew install postgresql
configure: error: libperl library is required for Perl

# postgresql depends on perl
❯❯❯ brew install perl

❯❯❯ brew install postgresql
/usr/bin/ld: /home/linuxbrew/.linuxbrew/lib/libuuid.a(gen_uuid.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC

# postgresql requires dynamically linked libuuid.so
❯❯❯ brew remove e2fsprogs
❯❯❯ brew install libuuid

❯❯❯ brew install postgresql

❯❯❯ brew install httpd24
/home/linuxbrew/.linuxbrew/Cellar/apr/1.5.2/libexec/build-1/libtool --silent --mode=link /usr/bin/gcc-4.8 -std=gnu99  -pthread  -Os -w -pipe -march=core2     -L/home/linuxbrew/.linuxbrew/lib -Wl,-rpath,/home/linuxbrew/.linuxbrew/lib    -o mod_authn_file.la -rpath NONE/libexec -module -avoid-version  mod_authn_file.lo 
libtool: link: only absolute run-paths are allowed
# prefix is NONE, for some odd reason.
```

**Comment by steowens** [comment]
This is old but I have the same problem.  here is the output of brew install:

``
WASE-AM02017840:php-5.6.11 sowens$ brew install berkeley-db4
==> Downloading https://homebrew.bintray.com/bottles/berkeley-db4-4.8.30.yosemit
######################################################################## 100.0%
==> Pouring berkeley-db4-4.8.30.yosemite.bottle.tar.gz
==> Caveats
This formula is keg-only, which means it was not symlinked into /usr/local.

BDB 4.8.30 is provided for software that doesn't compile against newer versions.

Generally there are no consequences of this for you. If you build your
own software and it requires this formula, you'll need to add to your
build variables:

    LDFLAGS:  -L/usr/local/opt/berkeley-db4/lib
    CPPFLAGS: -I/usr/local/opt/berkeley-db4/include

==> Summary
🍺  /usr/local/Cellar/berkeley-db4/4.8.30: 4591 files, 88M
``

**Comment by steowens** [comment]
How can we get Configure to look in these locations for the needed header files?

**Comment by sjackman** [comment]
You can `brew link --force berkeley-db4`

**Comment by s172262** [comment]
You might have to deploy `ndbm.h` which is included `gdbm-devel`.
There is `ndbm.h` in _#{MacOS.sdk_path}/usr_ on OSX.
